sdk/storage/azure-storage-blob-stress/src/main/java/com/azure/storage/blob/stress/utils/OriginalContent.java

```diff
@@ -41,68 +45,51 @@ public Mono<Void> setupBlob(BlobAsyncClient blobClient, long blobSize) {
         }
 
         this.blobSize = blobSize;
-        return Mono.using(() -> new CrcInputStream(BLOB_CONTENT_HEAD, blobSize),
-            data -> blobClient
-                .upload(BinaryData.fromStream(data))
-                .doFinally(i -> dataChecksum = data.getCrc()),
+        return Mono.using(
+                () -> new CrcInputStream(BLOB_CONTENT_HEAD, blobSize),
+                data -> blobClient
+                        .upload(BinaryData.fromStream(data, blobSize))
+                        .then(data.getContentInfo())
+                        .doOnSuccess(info -> dataChecksum = info.getCrc()),
```

Comment:
Thoughts on rather than using `doOnSuccess` which is a side-effects operator we put this into a `map` call?